.github/workflows/unit_tests.sh

```diff
@@ -43,9 +44,17 @@ for phase in "${PHASES[@]}"; do
                 export CC=clang
                 export CXX=clang++
             fi
-            meson --werror -Dtests=unsafe -Dslow-tests=true -Dfuzz-tests=true -Dman=true build
+            if [[ "$phase" = "RUN_GCC" ]]; then
+                # See FIXME below
+                MESON_ARGS+=(-Db_coverage=true)
+                (set +x; while :; do echo -ne "\n[WATCHDOG] $(date)\n"; sleep 30; done) &
+            fi
+            meson --werror -Dtests=unsafe -Dslow-tests=true -Dfuzz-tests=true -Dman=true "${MESON_ARGS[@]}" build
```

Comment:
I'm not sure the fuzz targets should be included in the coverage reports. They're kind of different from the unit tests in the sense that they aren't exactly used to make sure the code works as expected so they probably shouldn't be mixed (if the idea is to help to figure out what is covered by the unit tests and what else should be added eventually). To improve the coverage of the fuzz targets I think the OSS-Fuzz coverage reports at https://oss-fuzz.com/coverage-report/job/libfuzzer_asan_systemd/latest should be used instead.

Reply:
Well, it's still code paths that are covered no? When I have time I was planning to also add the integration tests (if possible).

Reply:
I wouldn't say they're covered. They're just relatively randomly triggered to make sure the code doesn't crash but we have no idea whether that code does what it's supposed to do. The bar isn't that high there :-)

Reply:
I mean many of the unit tests are also quite superficial, same for the integration tests

Reply:
I kind of hoped they weren't superficial :-)

Anyway, looking at https://github.com/coverallsapp/github-action/issues/93, I think it would probably be better to put coveralls in a separate workflow to make it easier to turn it off when something like that happens.

Reply:
I use coveralls in other projects, to be honest I've seen Github Actions itself down way more often...

Reply:
Fair enough. Though I think the "unit tests" workflow is more complicated than it should probably be. It isn't obvious that apart from compiling `systemd` and running the tests "RUN_GCC" also adds `-Db_coverage` and generate coverage data. I'm not sure whether anyone runs the script locally but as far as I can remember the idea was to make it relatively easy to build and test `systemd` by just running something like `unit_tests.sh SETUP RUN CLEANUP`.

Reply:
given the unsafe tests are ran, which break network connectivity on the running host as you found out, it's probably unwise to do that anyway

Reply:
I turned off all the additional tests including the unsafe ones in https://github.com/evverx/systemd/pull/7 and it didn't help much, which probably means that the "safe" tests are somehow equally unsafe in terms of breaking hosts they're run on :-) Either way, given that that approach appears to help to save some resources by mixing the tests and coverage I wouldn't say I'm opposed to it.

Reply:
[I'm a bit late to the party, so apologies if I'm repeating something that was already said.]

For coverage of fuzz tests, we shouldn't need to compile with `-Db_coverage`. Since f78ad5f0467de5066d5fed6c5c61bd740ba82e3b, we also compile and execute the fuzzer code without any sanitizers. 

I also don't think it's necessary/useful to enable slow-tests: the flag is used to enable additional loops in various cases, but this is not going to affect coverage in any noticable way. And if timeouts are a problem, then it'd be better to skip the slow tests here.

(`-Dfuzz-tests=true` is good, because this runs the fuzzer tests with the regression test cases, increasing coverage. It is also very quick.)

`-Dman` is also quite slow. It wouldn't be useful for coverage…

As for "safe" tests breaking networking: this shouldn't happen. I run the tests in VMs and containers over the network all the time, and haven't seen this… But if there's some combination or scenario where this happens, then we should figure it out.

Reply:
The thing is, to exclude these things another separate job would be needed instead of reusing the existing one, which means further slowing things down. The penalty for the coverage flags is ~1m 30s: this run took 8m 48s while it took 7m 10s to do this one without: https://github.com/systemd/systemd/pull/20846/checks?check_run_id=3714904033
Not sure it's worth a different job?

Reply:
Oh, OK. In that case no.

But I think it'd still make sense to at least do the man page only once.

Reply:
ok, restricted the manpage build to the clang run only (to share the load)